composer/datasets/evaluator.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import List
+
+import yahp as hp
+from torchmetrics.classification.accuracy import Accuracy
+from torchmetrics.classification.f_beta import F1
+from torchmetrics.classification.matthews_corrcoef import MatthewsCorrcoef
+from torchmetrics.collections import MetricCollection
+from torchmetrics.metric import Metric
+from torchmetrics.regression.pearson import PearsonCorrcoef
+from torchmetrics.regression.spearman import SpearmanCorrcoef
+
+from composer.core.types import Metrics
+from composer.datasets.dataset_registry import get_dataset_registry
+from composer.datasets.hparams import DataloaderSpec, DatasetHparams
+from composer.models.loss import CrossEntropyLoss
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class EvaluatorHparams(hp.Hparams):
+    hparams_registry = {  # type: ignore
+        "eval_dataset": get_dataset_registry(),
+    }
+
+    metric_registry = {
+        "Accuracy": Accuracy,
+        "MatthewsCorrcoef": MatthewsCorrcoef,
+        "F1": F1,
+        "PearsonCorrcoef": PearsonCorrcoef,
+        "SpearmanCorrcoef": SpearmanCorrcoef,
+        "CrossEntropyLoss": CrossEntropyLoss,
+    }
+
+    label: str = hp.required(doc="label")
+    eval_dataset: DatasetHparams = hp.required(doc="Training dataset hparams")
+    metrics: List[str] = hp.required(doc="metrics", template_default=list)
```

Comment:
Can we update the docs to describe if the metrics should be a valid instance of TorchMetrics?

Reply:
We should also add a validate() function that checks if the metrics are valid options in the registry ,and a helpful error message if not. Something like "The metric `KL-Divergence` was not found. Valid options are OPTION_1, OPTION_2, OPTION_3, etc"